src/bin/ion/commands/generate/model.rs

```diff
@@ -448,6 +455,58 @@ pub struct FieldReference(
     pub(crate) FieldPresence,
 );
 
+/// Represents an enum type
+/// e.g. Given below ISL,
+/// ```
+/// type::{
+///   name: enum_type,
+///   valid_values: [foo, bar, baz]
+/// }
+/// ```
+/// Corresponding generated code in Rust would look like following:
+/// ```
+/// enum EnumType {
+///    Foo,
+///    Bar,
+///    Baz
+/// }
+/// ```
+#[allow(dead_code)]
+#[derive(Debug, Clone, Builder, PartialEq, Serialize)]
+#[builder(setter(into))]
+pub struct Enum {
+    // Represents the fully qualified name for this data model
+    pub(crate) name: FullyQualifiedTypeName,
+    // Represents enum variants
+    variants: HashSet<EnumVariant>,
```

Comment:
I think that we should take an opinionated stance and say that enum variants must be defined using symbol values. If we follow through with that, then this can be changed to `HashSet<String>` (or even `&[String]`).

From what I can tell, `HashSet` does not have a stable iteration order. In some programming languages, changing the order of enum variants is not a backwards compatible change, so we need to ensure that the order we use is stable. I recommend using the order in which they are defined in the ISL. We could use a `LinkedHashSet` or a `BTreeSet`, but a `&[String]` is also just fine. (We expect the number of enum variants to be relatively small (`=< 100`) in computing terms, so an `O(n log n)` cost to check for duplicates isn't a big deal.)

Reply:
Using `BTreeSet` for preserving order of enum variants.